manifests/postfix_exporter.pp

```diff
@@ -0,0 +1,142 @@
+#
+#  [*extra_options*]
+#  Extra options added to the startup command
+#
+#  [*group*]
+#  Group under which the binary is running
+#
+#  [*init_style*]
+#  Service startup scripts style (e.g. rc, upstart or systemd)
+#
+#  [*install_method*]
+#  Installation method: url or package (only url is supported currently)
+#
+#  [*manage_group*]
+#  Whether to create a group for or rely on external code for that
+#
+#  [*manage_service*]
+#  Should puppet manage the service? (default true)
+#
+#  [*manage_user*]
+#  Whether to create user or rely on external code for that
+#
+#  [*os*]
+#  Operating system (linux is the only one supported)
+#
+#  [*package_ensure*]
+#  If package, then use this for package ensure default 'latest'
+#
+#  [*package_name*]
+#  The binary package name - not available yet
+#
+#  [*purge_config_dir*]
+#  Purge config files no longer generated by Puppet
+#
+#  [*restart_on_change*]
+#  Should puppet restart the service on configuration change? (default true)
+#
+#  [*service_enable*]
+#  Whether to enable the service from puppet (default true)
+#
+#  [*service_ensure*]
+#  State ensured for the service (default 'running')
+#
+#  [*user*]
+#  User which runs the service
+#
+#  [*version*]
+#  The binary release version
+
+class prometheus::postfix_exporter (
+  String $download_extension              = '',
+  Stdlib::HTTPUrl $download_url_base      = 'https://github.com/kumina/postfix_exporter/releases',
+  String[1] $group                        = 'postfix-exporter',
+  Array[String[1]] $extra_groups          = [],
+  String[1] $package_ensure               = 'installed',
+  String[1] $package_name                 = 'postfix_exporter',
+  String[1] $user                         = 'postfix-exporter',
+  String[1] $version                      = '0.2.0',
+  Boolean $purge_config_dir               = true,
+  Boolean $restart_on_change              = true,
+  Boolean $service_enable                 = true,
+  String[1] $service_ensure               = 'running',
+  String $service_name                    = 'postfix_exporter',
+  String[1] $init_style                   = $prometheus::init_style,
+  String[1] $install_method               = $prometheus::install_method,
+  Boolean $manage_group                   = true,
+  Boolean $manage_service                 = true,
+  Boolean $manage_user                    = true,
+  String[1] $os                           = $prometheus::os,
+  String $extra_options                   = '',
+  Hash[String, Scalar] $env_vars          = {},
+  Optional[Stdlib::HTTPUrl] $download_url = undef,
+  String[1] $config_mode                  = $prometheus::config_mode,
+  String[1] $arch                         = $prometheus::real_arch,
+  Stdlib::Absolutepath $bin_dir           = $prometheus::bin_dir,
+
+  Boolean        $export_scrape_job = false,
+  Stdlib::Port   $scrape_port       = 9154,
+  String[1]      $scrape_job_name   = 'postfix',
+  Optional[Hash] $scrape_job_labels = undef,
```

Comment:
there's some whitespace/alignment issue here...

```suggestion
  Stdlib::Absolutepath $bin_dir           = $prometheus::bin_dir,
  Boolean        $export_scrape_job       = false,
  Stdlib::Port   $scrape_port             = 9154,
  String[1]      $scrape_job_name         = 'postfix',
  Optional[Hash] $scrape_job_labels       = undef,
```

hmm... that commit suggestion doesn't show up properly, but i think you can figure out what i mean. :) update: see d565e4e

Reply:
That was actually on purpose.  Since data types came along, I've mostly given up on trying to align everything.  If it doesn't improve readability, why bother?  In this case, I've aligned the related scrape_job parameters and separated them from the others with a blank line.  Constantly realigning everything also creates a lot of noise.  Sure, when reviewing a PR, you can tell github to hide whitespace changes, but it doesn't give you an option to do that when using `blame` (although you can on the command-line I think).